File: app/src/main/java/nl/whitedove/agecalculator/CustomListAdapterDates.java
package nl.whitedove.agecalculator;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.List;

class CustomListAdapterDates extends BaseAdapter {

    private List<DatumGeval> listData;
    private LayoutInflater layoutInflater;

    CustomListAdapterDates(Context context, List<DatumGeval> listData) {
        this.listData = listData;
        layoutInflater = LayoutInflater.from(context);
    }

    @Override
    public int getCount() {
        return listData.size();
    }

    @Override
    public Object getItem(int position) {
        return listData.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    public View getView(int position, View convertView, ViewGroup parent) {

        ViewHolder holder;
        if (convertView == null) {
            convertView = layoutInflater.inflate(R.layout.dates_list_row, parent, false);
            holder = new ViewHolder();
            holder.tvWhat = convertView.findViewById(R.id.tvDrWhat);
            holder.tvWhenDate = convertView.findViewById(R.id.tvDrWhenDate);
            holder.tvWhenTime = convertView.findViewById(R.id.tvDrWhenTime);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }

        DatumGeval dat = listData.get(position);
        String sAantal = Helper.NumToString(dat.getAantal());
        String s = String.format("%s %ss", sAantal, dat.getEenheid());
        holder.tvWhat.setText(s);
        Helper.eenheidType eenh = dat.getEenheid();
        holder.tvWhenDate.setText(Helper.dFormat.print(dat.getDatumTijd()));
        if (eenh == Helper.eenheidType.hour || eenh == Helper.eenheidType.minute || eenh == Helper.eenheidType.second) {
            holder.tvWhenTime.setText(Helper.tFormatHhMmSs.print(dat.getDatumTijd()));
            holder.tvWhenTime.setVisibility(View.VISIBLE);
        } else {
            holder.tvWhenTime.setVisibility(View.GONE);
        }

        return convertView;
    }

    private static class ViewHolder {
        TextView tvWhat;
        TextView tvWhenDate;
        TextView tvWhenTime;
    }
}

File: app/src/main/java/nl/whitedove/agecalculator/CustomListAdapterDates.kt
package nl.whitedove.agecalculator

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import nl.whitedove.agecalculator.Helper.EenheidType

internal class CustomListAdapterDates(context: Context, private val listData: ArrayList<DatumGeval>) : BaseAdapter() {
    private val layoutInflater: LayoutInflater = LayoutInflater.from(context)
    override fun getCount(): Int {
        return listData.size
    }

    override fun getItem(position: Int): Any {
        return listData[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var cv = convertView
        val holder: ViewHolder
        if (cv == null) {
            cv = layoutInflater.inflate(R.layout.dates_list_row, parent, false)
            holder = ViewHolder()
            holder.tvWhat = cv!!.findViewById(R.id.tvDrWhat)
            holder.tvWhenDate = cv.findViewById(R.id.tvDrWhenDate)
            holder.tvWhenTime = cv.findViewById(R.id.tvDrWhenTime)
            cv.tag = holder
        } else {
            holder = cv.tag as ViewHolder
        }
        val dat = listData[position]
        val kind = dat.getKind()
        if (kind == Helper.KindType.Absolute) {

            val sAantal = Helper.numToString(dat.getAantal())
            val s = String.format("%s %ss", sAantal, dat.getEenheid())
            holder.tvWhat!!.text = s
        } else {
            val s = dat.getTextToShow()
            holder.tvWhat!!.text = s
        }

        val eenh = dat.getEenheid()
        holder.tvWhenDate!!.text = Helper.dFormat.print(dat.getDatumTijd())
        if (eenh == EenheidType.Hour || eenh == EenheidType.Minute || eenh == EenheidType.Second) {
            holder.tvWhenTime!!.text = Helper.tFormatHhMmSs.print(dat.getDatumTijd())
            holder.tvWhenTime!!.visibility = View.VISIBLE
        } else {
            holder.tvWhenTime!!.visibility = View.GONE
        }
        return cv
    }

    private class ViewHolder {
        var tvWhat: TextView? = null
        var tvWhenDate: TextView? = null
        var tvWhenTime: TextView? = null
    }

}